cms/models.py

```diff
@@ -40,17 +48,76 @@ class HomePage(Page):
         help_text="The subtitle text to display in the hero section of the home page.",
     )
 
+    product_section_title = models.CharField(
+        max_length=255,
+        null=True,
+        blank=True,
+        help_text="The title text to display in the product cards section of the home page.",
+    )
+
     content_panels = Page.content_panels + [
         ImageChooserPanel("hero"),
         FieldPanel("hero_title"),
         FieldPanel("hero_subtitle"),
+        FieldPanel("product_section_title"),
+        InlinePanel("featured_products", label="Featured Products"),
     ]
+
     parent_page_types = [Page]
     subpage_types = [
-        "CoursePage",
         "ResourcePage",
     ]
 
+    def _get_child_page_of_type(self, cls):
+        """Gets the first child page of the given type if it exists"""
+        child = self.get_children().type(cls).live().first()
+        return child.specific if child else None
+
+    @property
+    def products(self):
+        page_data = []
+        for page in self.featured_products.all():
+            if page.course_product_page:
+                product_page = page.course_product_page.specific
+                page_data.append(
+                    {
+                        "title": product_page.title,
+                        "description": product_page.description,
+                        "feature_image": product_page.feature_image,
+                        "start_date": product_page.product.first_unexpired_run.start_date,
```

Comment:
Sorry, I made a mistake in my suggestions here earlier. `first_unexpired_run` can return None, so this needs to be updated to handle that possiblity:

```suggestion
                product_page = page.course_product_page.specific
                run = product_page.product.first_unexpired_run
                page_data.append(
                    {
                        "title": product_page.title,
                        "description": product_page.description,
                        "feature_image": product_page.feature_image,
                        "start_date": run.start_date if run is not None else None,
```